lib/icinga/checkable-notification.cpp

```diff
@@ -145,35 +146,56 @@ static void FireSuppressedNotifications(Checkable* checkable)
 
 	int subtract = 0;
 
-	for (auto type : {NotificationProblem, NotificationRecovery, NotificationFlappingStart, NotificationFlappingEnd}) {
-		if (suppressed_types & type) {
-			bool still_applies = checkable->NotificationReasonApplies(type);
-
-			if (still_applies) {
-				bool still_suppressed;
-
-				switch (type) {
-					case NotificationProblem:
-						/* Fall through. */
-					case NotificationRecovery:
-						still_suppressed = !checkable->IsReachable(DependencyNotification) || checkable->IsInDowntime() || checkable->IsAcknowledged();
-						break;
-					case NotificationFlappingStart:
-						/* Fall through. */
-					case NotificationFlappingEnd:
-						still_suppressed = checkable->IsInDowntime();
-						break;
-					default:
-						break;
-				}
+	{
+		int_fast8_t wasLastParentRecoveryRecent = -1;
+
+		for (auto type : {NotificationProblem, NotificationRecovery, NotificationFlappingStart, NotificationFlappingEnd}) {
+			if (suppressed_types & type) {
+				bool still_applies = checkable->NotificationReasonApplies(type);
+
+				if (still_applies) {
+					bool still_suppressed;
+
+					switch (type) {
+						case NotificationProblem:
+							/* Fall through. */
+						case NotificationRecovery:
+							still_suppressed = !checkable->IsReachable(DependencyNotification) || checkable->IsInDowntime() || checkable->IsAcknowledged();
+							break;
+						case NotificationFlappingStart:
+							/* Fall through. */
+						case NotificationFlappingEnd:
+							still_suppressed = checkable->IsInDowntime();
+							break;
+						default:
+							break;
+					}
+
+					if (!still_suppressed && !checkable->IsLikelyToBeCheckedSoon()) {
+						if (wasLastParentRecoveryRecent < 0) {
+							auto threshold (Utility::GetTime() - 60);
```

Comment:
I'm wondering if it would be better to use the execution start time of the last check result instead as this would otherwise delay notifications by up to a minute. But then it wouldn't notify if the service is overdue, unsure if this is good or bad.

Reply:
The execution start time of the last check result doesn’t provide the expected behavior.

Reply:
What happens instead?

Reply:
I should ask this _you_ as _you_'ve suggested this.

Reply:
So I tried and it does not blow up in an obvious way when using `auto threshold (checkable->GetLastCheckResult()->GetExecutionStart());` instead.

Reply:
Sure, it does not blow up. It silences notifications until the next check.